codesign/lib/src/google_cloud_storage.dart

```diff
@@ -51,17 +41,16 @@ class EngineArtifactTransfer {
     int exitCode = 0,
   }) async {
     final String sourceUrl = '$gsCloudBaseUrl/flutter/$commitHash/$remotePath';
-    return await downloadFunction!(
```

Comment:
What's the value in having downloadFunction and downloadEngineArtifact?

nit: You don't need to include "Function" in the name of your methods

Reply:
umm right they come from an earlier review https://github.com/flutter/cocoon/pull/2125#discussion_r971240252 to have a wrapper class around the generic download function. I have reverted it back to its original state (without generic-wrapper).